### PR TITLE
Add gitflow versioning

### PR DESCRIPTION
**Pull request by yuriyshafranyuk1**
Added ability to make release with github release tags. Used https://github.com/aleksandr-m/gitflow-maven-plugin maven plugin for release.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -17,3 +18,14 @@ deployment:
...
     commands:
       - mvn deploy -DskipTests=true
+  release:
+      tag: /.*/
+      commands:
+        - git config user.name "circleci"
+        - git fetch
+        - git checkout $RELEASE_BRANCH
+        - git pull origin $RELEASE_BRANCH
+        - git reset --hard
```
Comment (yuriyshafranyuk1) [inline circle.yml:28]: some files changes during tests

File: circle.yml [hunk 2]
```diff
@@ -17,3 +18,14 @@ deployment:
...
+        - git tag -d $CIRCLE_TAG
```
Comment (yuriyshafranyuk1) [inline circle.yml:29]: Without this release fails. gitflow will create tag with same name. Its only on circleci machine without pushing

**Comment by yuriyshafranyuk1** [comment]
[ch304]

File: pom.xml [hunk 3]
```diff
@@ -519,6 +553,30 @@
...
+                <version>1.8.0</version>
+                <configuration>
+                    <verbose>true</verbose>
+                    <versionDigitToIncrement>2</versionDigitToIncrement>
+                    <releaseRebase>true</releaseRebase>
+                    <pushRemote>false</pushRemote>
+                    <skipTestProject>true</skipTestProject>
+                    <allowSnapshots>true</allowSnapshots>
+                    <gitFlowConfig>
+                        <productionBranch>versioning-test</productionBranch>
```
Comment (osklyarenko) [inline pom.xml]: bug

**Review by zdmytriv — APPROVED** [review]

**Pull request merged**